catalogue/webapp/components/IIIFViewer/MainViewer.tsx

```diff
@@ -191,6 +188,11 @@ const ItemRenderer = memo(({ style, index, data }: ItemRendererProps) => {
     OverlayPositionData[]
   >([]);
 
+  useEffect(async () => {
+    const ocrText = await getCanvasOcr(canvases[index]);
+    setOcrText(ocrText || missingAltTextMessage);
+  }, [index]);
```

Comment:
Does this give us the `useEffect function must return a cleanup function or nothing` warning? As the `useEffect` callback shouldn't return a Promise we would [usually](https://www.robinwieruch.de/react-hooks-fetch-data/) do something like
```typescript
useEffect(() => {
  const fetchOcr = async () => {
    const ocrText = await getCanvasOcr(canvases[index]);
    setOcrText(ocrText || missingAltTextMessage);
  }
  fetchOcr();
}, [index]);
```

Reply:
ah, good point